### PR TITLE
feat: Add AnnotationsService

### PR DESCRIPTION
**Pull request by bahlo**
This adds support for Annotations: https://axiom.co/docs/restapi/endpoints/getAnnotations

### CODE REVIEW

File: axiom/annotations.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+// AnnotationsFilter filters annotations on the [`List`] method.
+type AnnotationsFilter struct {
+	Datasets []string   `url:"datasets"`
+	Start    *time.Time `url:"start"`
+	End      *time.Time `url:"start"`
+}
+
+// List annotations.
+func (a *AnnotationsService) List(ctx context.Context, filter *AnnotationsFilter) ([]*Annotation, error) {
+	ctx, span := a.client.trace(ctx, "Annotations.List", trace.WithAttributes())
```
Comment (lukasmalkmus) [inline axiom/annotations.go:102]: Please add attributes to `trace.WithAttributes()` or drop it. Ideally, add them conditionally if `filter != nil`.

File: axiom/annotations.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	var res Annotation
+	if err := a.client.Call(ctx, http.MethodGet, path, nil, &res); err != nil {
+		return nil, spanError(span, err)
+	}
+
+	return &res, nil
+}
+
+func (a *AnnotationsService) Update(ctx context.Context, id string, annotation *AnnotationUpdateRequest) (*Annotation, error) {
+	ctx, span := a.client.trace(ctx, "Annotations.Update")
```
Comment (lukasmalkmus) [inline axiom/annotations.go:139]: Add annotation ID as attribute.

File: axiom/annotations.go (hunk 1 above)
Comment (bahlo) [reply axiom/annotations.go]: Fixed in 94f6afd

File: axiom/annotations.go (hunk 2 above)
Comment (bahlo) [reply axiom/annotations.go]: Fixed in 94f6afd

File: axiom/annotations.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+	var res Annotation
+	if err := a.client.Call(ctx, http.MethodPost, a.basePath, annotation, &res); err != nil {
+		return nil, spanError(span, err)
+	}
+
+	return &res, nil
+}
+
+// AnnotationsFilter filters annotations on the [`List`] method.
```
Comment (lukasmalkmus) [inline axiom/annotations.go:93]:
```
[`List`]
```

This is odd syntax. To render this properly in GoDoc, please do `[AnnotationsService.List]`.

File: axiom/annotations.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+type AnnotationsFilter struct {
+	Datasets []string   `url:"datasets"`
+	Start    *time.Time `url:"start"`
+	End      *time.Time `url:"start"`
```
Comment (lukasmalkmus) [inline axiom/annotations.go:97]: I don't think those should be pointers. Pointers to times are annoying to use from a users perspective.

Comment (bahlo) [reply axiom/annotations.go]: 84efb90

File: axiom/annotations.go (hunk 3 above)
Comment (bahlo) [reply axiom/annotations.go]: 84efb90

File: axiom/annotations.go (hunk 4 above)
Comment (bahlo) [reply axiom/annotations.go]: f6d0be8

File: axiom/annotations.go (hunk 3 above)
Comment (bahlo) [reply axiom/annotations.go]: f6d0be8

**Review by lukasmalkmus — APPROVED** [review]

**Pull request merged**